src/test/java/io/odpf/firehose/sink/prometheus/PromSinkTest.java

```diff
@@ -0,0 +1,312 @@
+import io.odpf.firehose.exception.NeedToRetry;
+import io.odpf.firehose.metrics.Instrumentation;
+import io.odpf.firehose.sink.prometheus.request.PromRequest;
+import org.apache.http.Header;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.StatusLine;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpEntityEnclosingRequestBase;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.message.BasicHeader;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.xerial.snappy.Snappy;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static io.odpf.firehose.sink.prometheus.PromSinkConstants.PROMETHEUS_LABEL_FOR_METRIC_NAME;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.Mockito.*;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+@RunWith(MockitoJUnitRunner.class)
+public class PromSinkTest {
+    @Mock
+    private Instrumentation instrumentation;
+    @Mock
+    private PromRequest request;
+    @Mock
+    private HttpClient httpClient;
+    @Mock
+    private StencilClient stencilClient;
+    @Mock
+    private HttpPost httpPost;
+    @Mock
+    private HttpResponse response;
+    @Mock
+    private HttpEntity httpEntity;
+    @Mock
+    private StatusLine statusLine;
+    @Mock
+    private Map<Integer, Boolean> retryStatusCodeRange;
+    @Mock
+    private Map<Integer, Boolean> requestLogStatusCodeRanges;
+
+    private List<Message> messages;
+    private Cortex.WriteRequest writeRequest;
+    private List<HttpEntityEnclosingRequestBase> httpPostList;
+
+    @Before
+    public void setup() {
+        initMocks(this);
+        httpPostList = Collections.singletonList(httpPost);
+
+        Cortex.Sample sample = Cortex.Sample.newBuilder().setValue(10).setTimestampMs(1000000).build();
+        Cortex.LabelPair labelPair = Cortex.LabelPair.newBuilder().setName(PROMETHEUS_LABEL_FOR_METRIC_NAME).setValue("test_metric").build();
+        Cortex.TimeSeries timeSeries = Cortex.TimeSeries.newBuilder()
+                .addSamples(sample).addLabels(labelPair)
+                .build();
+        writeRequest = Cortex.WriteRequest.newBuilder().addTimeseries(timeSeries).build();
+
+        messages = new ArrayList<>();
+        Message message = new Message(null, writeRequest.toByteArray(), "", 0, 1);
+
+        messages.add(message);
+    }
+
+    @Test
+    public void shouldPrepareRequestDuringPreparationAndCallItDuringExecution() throws Exception {
```

Comment:
the name and intent of the tests are different. Can we name it shouldCallHttpClientWithProperRequest